### ISSUE TITLE
`Lint/RedundantRequireStatement` breaks code with modifiers

### ISSUE THREAD

**Issue by tsugimoto** [rubocop/rubocop#11120]
```ruby
require 'pp' unless respond_to?(:pretty_inspect)
```

Autocorrect breaks code when `require` is followed by postfix `unless`.

--------

## Expected behavior

Autocorrects into valid Ruby code, possibly like below (considering the modifier condition may have side effects.)

```ruby
respond_to?(:pretty_inspect)
```

## Actual behavior

```
Offenses:

main.rb:1:1: W: [Correctable] Lint/RedundantRequireStatement: Remove unnecessary require statement.
require 'pp' unless respond_to?(:pretty_inspect)
^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

effectively autocorrects code into

```ruby
unless respond_to?(:pretty_inspect)
```

, which is syntactically incorrect.

## Steps to reproduce the problem

```
$ cat main.rb
require 'pp' unless respond_to?(:pretty_inspect)
$ rubocop --only Lint/RedundantRequireStatement
```

## RuboCop version

```
$ rubocop -V
1.37.1 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.1) [x86_64-darwin20]
$ 
```
